include/picongpu/fields/incidentField/profiles/GaussianBeam.def

```diff
@@ -46,81 +48,32 @@ namespace picongpu
-                         *  unit: seconds (1 sigma) */
-                        static constexpr float_64 PULSE_LENGTH_SI = 10.615e-15 / 4.0;
-
-                        /** beam waist: distance from the axis where the pulse intensity (E^2)
+                        /** Beam waist: distance from the axis where the pulse intensity (E^2)
                          *              decreases to its 1/e^2-th part,
                          *              at the focus position of the laser
                          * W0_SI = FWHM_of_Intensity / sqrt{ 2* ln(2) }
                          *                             [   1.17741    ]
                          *
-                         *  unit: meter */
+                         * unit: meter
+                         */
                         static constexpr float_64 W0_SI = 5.0e-6 / 1.17741;
-                        /** the distance to the laser focus in propagation direction
-                         *  unit: meter */
-                        static constexpr float_64 FOCUS_POS_SI = 4.62e-5;
 
                         /** The laser pulse will be initialized PULSE_INIT times of the PULSE_LENGTH
                          *
-                         *  unit: none */
+                         *  unit: none
+                         */
                         static constexpr float_64 PULSE_INIT = 20.0;
 
-                        /** laser phase shift (no shift: 0.0)
+                        /** Laguerre mode parameters
                          *
-                         * sin(omega*time + laser_phase): starts with phase=0 at center --> E-field=0 at center
-                         *
-                         * unit: rad, periodic in 2*pi
+                         * @{
                          */
-                        static constexpr float_X LASER_PHASE = 0.0;
-
                         using LAGUERREMODES_t = gaussianBeam::LAGUERREMODES_t;
                         using LAGUERREPHASES_t = gaussianBeam::LAGUERREPHASES_t;
                         static constexpr uint32_t MODENUMBER = gaussianBeam::MODENUMBER;
-
-                        /** Available E polarisation types, B polarization will be calculated automatically
-                         *
-                         * For linear polarization, AXIS_1 is next axis after the propagation axis in order (x, y, z)
-                         * with a periodic wrap. AXIS_2 is next after AXIS_1. E.g. for y propagation axis, AXIS_1 = z,
-                         * AXIS_2 = x.
-                         */
-                        enum PolarisationType
-                        {
-                            LINEAR_AXIS_1 = 1u,
-                            LINEAR_AXIS_2 = 2u,
-                            CIRCULAR = 4u,
-                        };
-                        /** Polarization selection
-                         */
-                        static constexpr PolarisationType Polarisation = CIRCULAR;
+                        /** @} */

```

Comment:
A lot of the parameters, like wavelength, pulse duration and a0, vanished and went to into BaseParam. I am not sure this is a good idea, because a user will need to copy all the parameters from two different files into one struct in his example. This is quite a complex a task, as the user would, first, need to know in in which files to look, and second, need to know where these are located.

In general, we currently miss an example showing how to configure a laser from a predefined profile. which together with the above makes this task impossible for all who do not know how to do it beforehand.

However, it somehow does make sense to define a base class for the parameters in common to all profiles.
I do not know how to solve this issue at the moment.

Reply:
This is true, but I do not think it's reasonably possible to keep lasers generalized otherwise. The old lasers are all largely copy-paste of one another, and I really felt it when first making their incident field analogs with doing exactly the same transformation over and over. It also makes a really bad convention that e.g. amplitude is always called the same, but it's never formalized as each parameter class is technically independent.

Wrt tp navigating, this should be solvable with doxygen. An example would also help of course

Reply:
Regarding examples, the plan is to transform most existing examples to use the incident field laser.

Reply:
Well, I think for now (until we transform  the examples, but possibly also after this), can you add a comment to the section in `incidentField.param`, where the predefined profiles are mentioned, which explains how the pre-defined profiles are configured. That is, saying that profiles are configured via a param struct which can be constructed by taking the base struct + the laser specific struct, as well as mentioning the place where to find these. Also we should probably add a comment there that it would be fail-safe practice to add the laser source to all incidentField planes, except YMax in moving window simulations (while making sure that the laser does not propagate along -y for these kind of simulations).

Also Fabia (new student) could change the laser initialization method in at least one of the examples directly after this PR is merged to practice incidentField setup, before implementing new profiles.

Reply:
Added a note about it to `incidentField.param`.